pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -67,11 +74,24 @@ func NewHookExecutor(client kclient.PodsNamespacer, tags client.ImageStreamTagsN
 // Execute executes hook in the context of deployment. The suffix is used to
 // distinguish the kind of hook (e.g. pre, post).
 func (e *HookExecutor) Execute(hook *deployapi.LifecycleHook, deployment *kapi.ReplicationController, suffix, label string) error {
-	var err error
+	config, err := deployutil.DecodeDeploymentConfig(deployment, e.decoder)
+	if err != nil {
+		return err
```

Comment:
Instead of bailing out you could potentially fallback to emitting on the replication controller (that's what we do in the deployment controller) but I don't feel really strong about it.